discovery/sync_manager_test.go

```diff
@@ -219,6 +219,41 @@ func TestSyncManagerInitialHistoricalSync(t *testing.T) {
 	assertNoMsgSent(t, extraPeer)
 }
 
+// TestSyncManagerHistoricalSyncOnReconnect tests that the sync manager will
+// re-trigger a historical sync when a new peer connects after a historical
+// sync has completed, but we have lost all peers.
+func TestSyncManagerHistoricalSyncOnReconnect(t *testing.T) {
```

Comment:
Nit: would prefer if this commit was squashed with what introduced the behavior.

Reply:
I prefer splitting a bugfix and the test reproducing the bug in separate commits. Especially when I'm reviewing this makes it easy to cherry-pick the test commit on master to exercise the failing scenario.

Reply:
Sure but that wasn't the case here. The non-test changes alone didn't have an impact on the tests 😛